### DISCUSSION TITLE
Assistance Needed: Firefly III Data Importer - "Configuration/Connection Error"

### DISCUSSION THREAD

**Discussion by Stealth0059** [firefly-iii/firefly-iii#9591] [Firefly III]
Hello everyone,

I'm having trouble getting the Firefly III Data Importer (v1.5.7) running on my setup. I followed the guide in the [documentation](https://docs.firefly-iii.org/how-to/data-importer/installation/self-managed/), but when I access the importer, I encounter the following error:

    "Configuration / connection error :("

Additionally, the web page is not formatted correctly, as shown in the attached screenshot:
![grafik](https://github.com/user-attachments/assets/bf2f9487-bc15-4a4b-8d4e-a9ea3c0c6776)

My Setup:
- Hardware: Raspberry Pi 4
- OS: Ubuntu 22.04.5 LTS
- Webserver: nginx 1.27.3
- PHP 8.3
- Firefly III Version 6.0.20
- Network:
  - The server is behind a router with HTTP and HTTPS ports routed to the Raspberry Pi.
  - There is another router in this network behind which lies my home network.
  - Firwall open Ports are 22, 80, 443

My observations:
- In the first log file, I noticed a 401 Authorization required error. However, this error never appeared again in subsequent log files.
   - During this time, I tried disabling authorization for the Firefly III installation, but I still didn’t see any new entries in the log file.
- When testing now, I no longer encounter the 401 error, but I also don’t see any helpful log entries to diagnose the issue.
- Every time I click the "Start Over" button in the Data Importer GUI, a new session identifier is successfully generated. This indicates that at least some functionality is working as expected.

I've attached the following for reference:
- My .env configuration for the Data Importer.
- My nginx configuration.
- Log files from the Data Importer’s storage/log folder

[env.txt](https://github.com/user-attachments/files/18276064/env.txt)
[HttpGateway.conf.txt](https://github.com/user-attachments/files/18276076/HttpGateway.conf.txt)
[fireflyimporter.mydomain.com.conf.txt](https://github.com/user-attachments/files/18276077/fireflyimporter.mydomain.com.conf.txt)
[data-import-2024-12-26.log.txt](https://github.com/user-attachments/files/18276092/data-import-2024-12-26.log.txt)
[data-import-2024-12-30.log.txt](https://github.com/user-attachments/files/18276075/data-import-2024-12-30.log.txt)

I suspect the issue might be related to either the configuration or the network setup. I couldn’t find any explicit port requirements in the documentation, which might also be a factor.

Any insights or guidance to resolve this would be greatly appreciated!

Thanks in advance for your help!

Best regards,

**Comment by JC5** [comment]
I see the CSS and JS are not working, are there any errors in your browser console?

**Reply by Stealth0059** [reply]
I checked the browser console and here are the errors I can see

    Console Tab:
		The resource from “https://fireflyimporter.mydomain.com/build/assets/app-amkGEgGS.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff). 
		Loading module from “https://fireflyimporter.mydomain.com/build/assets/index-CXBe0dho.js” was blocked because of a disallowed MIME type (“text/html”). 
		Loading failed for the module with source “https://fireflyimporter.mydomain.com/build/assets/index-CXBe0dho.js”. fireflyimporter.mydomain.com:205:399
		Loading module from “https://fireflyimporter.mydomain.com/build/assets/bootstrap-CJTxePce.js” was blocked because of a disallowed MIME type (“text/html”).
		Loading module from “https://fireflyimporter.mydomain.com/build/assets/vendor-DCxVLCSx.js” was blocked because of a disallowed MIME type (“text/html”). 
		The resource from “https://fireflyimporter.mydomain.com/build/assets/app-amkGEgGS.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).

    Network Tab:
        I noticed NS_ERROR_CORRUPTED_CONTENT errors in the network tab for the following files:
            app-amkGEgGS.css
            index-CXBe0dho.js
            bootstrap-CJTxePce.js
            vendor-DCxVLCSx.js

**Reply by JC5** [reply]
So it looks like the CSS and JS aren't loaded correctly. What is your configuration like?

**Reply by JC5** [reply]
Wait you shared that already lol. Let me check it out later today.

**Reply by JC5** [reply]
```
location ~ ^/(?:build|tests|config|lib|3rdparty|templates|data)(?:$|/)  { return 404; }
```

Remove this.

**Reply by Stealth0059** [reply]
Thanks for the suggestion! The issue was indeed related to the nginx configuration. After removing build from the blocked paths and making a few more adjustments to the Firefly III nginx config (htaccess ), along with updating Firefly III to the latest version, everything works now.

Thanks for the help!
